src/Lykke.Job.Messages/Sagas/BlockchainOperationsSaga.cs

```
@@ -40,45 +45,38 @@ public async Task Handle(BlockchainCashinDetector.Contract.Events.CashinComplete
             await SendCashinEmailAsync(evt.OperationId, evt.ClientId, evt.Amount, evt.AssetId, commandSender);
         }
 
-        //From CashoutProcessor
+        #region CashoutProcessor
+
         [UsedImplicitly]
-        public async Task Handle(CashinCompletedEvent evt, ICommandSender commandSender)
+        public async Task Handle(CrossClientCashoutCompletedEvent evt, ICommandSender commandSender)
         {
             await SendCashinEmailAsync(evt.OperationId, evt.ClientId, evt.Amount, evt.AssetId, commandSender);
         }
 
-        //From CashoutProcessor
         [UsedImplicitly]
         public async Task Handle(CashoutCompletedEvent evt, ICommandSender commandSender)
         {
-            var operationId = evt.OperationId;
-            if (await _deduplicationRepository.IsExistsAsync(operationId))
-                return;
-
-            var clientModel = await _clientAccountClient.GetByIdAsync(evt.ClientId.ToString());            
-            var asset = await _cachedAssetsService.TryGetAssetAsync(evt.AssetId);
+            await SendCashoutEmailAsync(evt.OperationId, evt.ClientId, evt.Amount, evt.AssetId, commandSender);
+        }
 
-            var parameters = new
+        [UsedImplicitly]
+        public async Task Handle(CashoutsBatchCompletedEvent evt, ICommandSender commandSender)
+        {
+            if (evt.Cashouts == null || evt.Cashouts.Length == 0)
             {
-                AssetId = asset.Id == LykkeConstants.LykkeAssetId ? EmailResources.LykkeCoins_name : asset.DisplayId,
-                Amount = NumberFormatter.FormatNumber(evt.Amount, asset.Accuracy),
-                ExplorerUrl = "",
-                //{ "ExplorerUrl", string.Format(_blockchainSettings.ExplorerUrl, messageData.SrcBlockchainHash },
-                Year = DateTime.UtcNow.Year.ToString()
-            };
+                _log.Warning($"Batch cashouts are empty. BatchId {evt.BatchId}", context: evt);
```

Better to throw an exception